ldapAdapterEXT-ext-impl/src/main/java/com/bosch/osmi/sw360/ldapAdapter/CustomPortalLDAPImporterImpl.java

```diff
@@ -131,5 +164,80 @@ private Organization addOrganization(String organizationName, User user) throws
 		}
         return organization;
 	}
+
+	private String mapOrganizationName(String name) {
+		loadLDAPImporterProperties();
+		String defaultTarget = defaultToNull ? null : name;
+
+		if (matchPrefix) {
+			return organizationMappings.keySet().stream()
+					.filter(name::startsWith)
+					.sorted(Comparator.comparing(String::length).reversed())
+					.findFirst()
+					.map(organizationMappings::get)
+					.orElse(defaultTarget);
+		} else { // match complete name
+			return organizationMappings.getOrDefault(name, defaultTarget);
+		}
+	}
+
+	private void loadLDAPImporterProperties() {
+		Properties ldapConfigProperties = loadProperties(CustomPortalLDAPImporterImpl.class, PROPERTIES_FILE_PATH);
+		matchPrefix = Boolean.parseBoolean(ldapConfigProperties.getProperty(MATCH_PREFIX_KEY, "false"));
+		defaultToNull = Boolean.parseBoolean(ldapConfigProperties.getProperty(DEFAULT_TO_NULL_KEY, "false"));
+
+		List<Object> mappingSourceKeys = ldapConfigProperties
+				.keySet()
+				.stream()
+				.filter(p -> ((String) p).startsWith(MAPPING_KEYS_PREFIX) && !((String) p).endsWith(MAPPING_VALUES_SUFFIX))
+				.collect(Collectors.toList());
+
+		organizationMappings = new HashMap<>();
+		for (Object sourceKey : mappingSourceKeys) {
+			String sourceOrg = ldapConfigProperties.getProperty((String)sourceKey);
+			String targetOrg = ldapConfigProperties.getProperty(sourceKey+MAPPING_VALUES_SUFFIX);
+			if (sourceOrg != null && targetOrg != null && sourceOrg.length() > 0 && targetOrg.length() > 0) {
+				organizationMappings.put(sourceOrg, targetOrg);
+			}
+		}
+	}
+
+	/**
+	 * Copied from https://github.com/sw360/sw360portal/blob/master/libraries/lib-datahandler/src/main/java/com/siemens/sw360/datahandler/common/CommonUtils.java
+	 */
+	private static Properties loadProperties(Class<?> clazz, String propertiesFilePath) {
+		return loadProperties(clazz, propertiesFilePath, true);
+	}
+
+	/**
+	 * Copied from https://github.com/sw360/sw360portal/blob/master/libraries/lib-datahandler/src/main/java/com/siemens/sw360/datahandler/common/CommonUtils.java
+	 */
+	private static Properties loadProperties(Class<?> clazz, String propertiesFilePath, boolean useSystemConfig) {
```

Comment:
You might be able to simplify this call:
- the `clazz` argument is always `CustomPortalLDAPImporterImpl.class`
- the `propertiesFilePath` is always `PROPERTIES_FILE_PATH`
- the `useSystemConfig` flag is always `true`